### ISSUE TITLE
Learning Rate will loop to 9

### ISSUE THREAD

**Issue by Night1099** [CGLiWang/DeepBasis_SVBRDF#1]
```

2024-05-15 14:05:19,043 INFO: [DeepB..][epoch:  2, iter: 200,300, lr:(1.003e-05,)] [eta: 11:17:51, time (data): 0.187 (0.000)] l_pix: 7.2781e-02 l_kesi: 6.1227e-03 
2024-05-15 14:05:39,213 INFO: [DeepB..][epoch:  2, iter: 200,400, lr:(1.002e-05,)] [eta: 11:17:30, time (data): 0.183 (0.000)] l_pix: 5.7615e-02 l_kesi: 4.9227e-03 
2024-05-15 14:05:57,627 INFO: [DeepB..][epoch:  2, iter: 200,500, lr:(1.001e-05,)] [eta: 11:17:08, time (data): 0.186 (0.000)] l_pix: 9.3073e-02 l_kesi: 6.5995e-03 
2024-05-15 14:06:16,131 INFO: [DeepB..][epoch:  2, iter: 200,600, lr:(1.000e-05,)] [eta: 11:16:45, time (data): 0.183 (0.000)] l_pix: 7.7909e-02 l_kesi: 5.9875e-03 
2024-05-15 14:06:34,586 INFO: [DeepB..][epoch:  2, iter: 200,700, lr:(9.995e-06,)] [eta: 11:16:23, time (data): 0.182 (0.000)] l_pix: 7.2582e-02 l_kesi: 6.4358e-03 
2024-05-15 14:06:53,026 INFO: [DeepB..][epoch:  2, iter: 200,800, lr:(9.988e-06,)] [eta: 11:16:01, time (data): 0.202 (0.001)] l_pix: 2.8512e-02 l_kesi: 4.9581e-03 
2024-05-15 14:07:13,689 INFO: [DeepB..][epoch:  2, iter: 200,900, lr:(9.980e-06,)] [eta: 11:15:41, time (data): 0.207 (0.001)] l_pix: 6.8267e-02 l_kesi: 6.5742e-03 

```
Once it gets past about 50000 steps the learning rate will drop below 1 and loop to 9 which explodes the model

I set it to a static 2e-05 learning rate and did training again, it makes really good progress till it gets done with first epoch then the next validation step all the maps turn into their blank colors seemingly like a light switch any reason for this?

im doing about a 167,000 material dataset

**Comment by Night1099** [comment]
![105000](https://github.com/CGLiWang/DeepBasis_SVBRDF/assets/90132896/7e5909a9-86f1-4078-b805-c848992a8601)

as you can see above its making good progress

then 5000 steps later

![110000](https://github.com/CGLiWang/DeepBasis_SVBRDF/assets/90132896/903afb4e-380e-46eb-9dea-c94849699728)

**Comment by CGLiWang** [comment]
> ```
> 
> 2024-05-15 14:05:19,043 INFO: [DeepB..][epoch:  2, iter: 200,300, lr:(1.003e-05,)] [eta: 11:17:51, time (data): 0.187 (0.000)] l_pix: 7.2781e-02 l_kesi: 6.1227e-03 
> 2024-05-15 14:05:39,213 INFO: [DeepB..][epoch:  2, iter: 200,400, lr:(1.002e-05,)] [eta: 11:17:30, time (data): 0.183 (0.000)] l_pix: 5.7615e-02 l_kesi: 4.9227e-03 
> 2024-05-15 14:05:57,627 INFO: [DeepB..][epoch:  2, iter: 200,500, lr:(1.001e-05,)] [eta: 11:17:08, time (data): 0.186 (0.000)] l_pix: 9.3073e-02 l_kesi: 6.5995e-03 
> 2024-05-15 14:06:16,131 INFO: [DeepB..][epoch:  2, iter: 200,600, lr:(1.000e-05,)] [eta: 11:16:45, time (data): 0.183 (0.000)] l_pix: 7.7909e-02 l_kesi: 5.9875e-03 
> 2024-05-15 14:06:34,586 INFO: [DeepB..][epoch:  2, iter: 200,700, lr:(9.995e-06,)] [eta: 11:16:23, time (data): 0.182 (0.000)] l_pix: 7.2582e-02 l_kesi: 6.4358e-03 
> 2024-05-15 14:06:53,026 INFO: [DeepB..][epoch:  2, iter: 200,800, lr:(9.988e-06,)] [eta: 11:16:01, time (data): 0.202 (0.001)] l_pix: 2.8512e-02 l_kesi: 4.9581e-03 
> 2024-05-15 14:07:13,689 INFO: [DeepB..][epoch:  2, iter: 200,900, lr:(9.980e-06,)] [eta: 11:15:41, time (data): 0.207 (0.001)] l_pix: 6.8267e-02 l_kesi: 6.5742e-03 
> ```
> 
> Once it gets past about 50000 steps the learning rate will drop below 1 and loop to 9 which explodes the model
> 
> I set it to a static 2e-05 learning rate and did training again, it makes really good progress till it gets done with first epoch then the next validation step all the maps turn into their blank colors seemingly like a light switch any reason for this?
> 
> im doing about a 167,000 material dataset

What is your meaning of "the learning rate will drop below 1 and loop to 9"?  According to your post log, from 1.003e-05 to  9.980e-06 is a correct declaying.

**Comment by CGLiWang** [comment]
> all the maps turn into their blank colors

The variation of maps seems not bad, and i don't understand

> ![105000](https://private-user-images.githubusercontent.com/90132896/331341739-7e5909a9-86f1-4078-b805-c848992a8601.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5MDQxMDgsIm5iZiI6MTcxNTkwMzgwOCwicGF0aCI6Ii85MDEzMjg5Ni8zMzEzNDE3MzktN2U1OTA5YTktODZmMS00MDc4LWI4MDUtYzg0ODk5MmE4NjAxLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE2VDIzNTY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3MWZiYmFjZDhjMWQ0ZDk0NjBlN2NkMTRiMzFiZmU5ZWZkMDc1ZWU4NmU4NmFkYjRjYTdiZGZhMzMxNDRlYjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DYjldjwL4pRc1ecAYfEanr5gU1bl1RzVrPzOh83qDEs)
> 
> as you can see above its making good progress
> 
> then 5000 steps later
> 
> ![110000](https://private-user-images.githubusercontent.com/90132896/331341837-903afb4e-380e-46eb-9dea-c94849699728.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5MDQxMDgsIm5iZiI6MTcxNTkwMzgwOCwicGF0aCI6Ii85MDEzMjg5Ni8zMzEzNDE4MzctOTAzYWZiNGUtMzgwZS00NmViLTlkZWEtYzk0ODQ5Njk5NzI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE2VDIzNTY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhOTNjNzgyMDdkMjNlZjEwZTRkOTY5YjJlYWRmMjNmMWM2ODQ0NmU0ODA4MGYzMzQ1ODkxN2I1N2YzODdjNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.foaY2f_Fh4wyvA6qLMvO4QIgfivoemF5l2DdtBZuDuM)

I did not observe the issue you mentioned, "all the maps turn into their blank colors." The changes in these two images seem to be normal.

**Comment by Night1099** [comment]
Oh my bad the learning rate is correct i read it wrong

but youre saying the maps seemingly resetting is a normal thing and i should wait for it to train longer?


im talking about this


![110000](https://github.com/CGLiWang/DeepBasis_SVBRDF/assets/90132896/f3c6eabe-2112-42a4-876f-619da5643fd8)


to this within 5000 steps


![2](https://github.com/CGLiWang/DeepBasis_SVBRDF/assets/90132896/1df7af91-00b9-4838-b3d0-1bd64cf7cfdb)

**Comment by CGLiWang** [comment]
> Oh my bad the learning rate is correct i read it wrong
> 
> but youre saying the maps seemingly resetting is a normal thing and i should wait for it to train longer?
> 
> im talking about this
> 
> ![110000](https://private-user-images.githubusercontent.com/90132896/331425567-f3c6eabe-2112-42a4-876f-619da5643fd8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5MDk0NjUsIm5iZiI6MTcxNTkwOTE2NSwicGF0aCI6Ii85MDEzMjg5Ni8zMzE0MjU1NjctZjNjNmVhYmUtMjExMi00MmE0LTg3NmYtNjE5ZGE1NjQzZmQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE3VDAxMjYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0OWQ5MDlkZWU0NGFkYzRjN2Y1MTFkODBjMjRjYWJjMjI2ZDc3OWUzZDFjYzc3ZjA5YTkxM2Q1YjZkNjA4YTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fpXkVd7xm2ePJaQPh_59ZwU1n1Xo10WfuvQ4yU_XmKE)
> 
> to this within 5000 steps
> 
> ![2](https://private-user-images.githubusercontent.com/90132896/331425577-1df7af91-00b9-4838-b3d0-1bd64cf7cfdb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5MDk0NjUsIm5iZiI6MTcxNTkwOTE2NSwicGF0aCI6Ii85MDEzMjg5Ni8zMzE0MjU1NzctMWRmN2FmOTEtMDBiOS00ODM4LWIzZDAtMWJkNjRjZjdjZmRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE3VDAxMjYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlMjcyNDQyMjY4NWE2NjhhY2JkM2JlYzk0NWMzZmQxZjUyMDM2MmVjNjIwNzE5NDJkY2Q0OWMzZWQ1ODE3OGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3CUJl7SH732sPWv1_zXZ_8Wt4fxK453axwwG7pKheTI)

Yeah, I see the details of normal are lost, but no artifacts appears. It is maybe depend on your task/dataset/or something to decide whether need longer training.

**Comment by CGLiWang** [comment]
A suggestion: you can use a tensorboard to visualize the trend of loss decaying, which is a good tool to aid in training time decisions.

**Comment by Night1099** [comment]
Ok i will try again thanks

**Comment by Night1099** [comment]
Ok update on this

![image](https://github.com/CGLiWang/DeepBasis_SVBRDF/assets/90132896/10bb1bd3-4e2c-4ea5-864d-3d09c7c4f58b)


the huge spike at around 60k is the moment the detail in normal is lost

and then the loss start to come back down



this is step 285,000, still no detail in normal

![285000](https://github.com/CGLiWang/DeepBasis_SVBRDF/assets/90132896/5e28684b-0742-440e-b31b-d3e7d77acb71)


this is my train command

```
python train.py --name DeepBasisTraining7 --save_root ./output --train_data_root ./dataset/train --test_data_root ./dataset/test --use_tb_logger --total_iter 1000000 --print_freq 100 --save_freq 50000 --test_freq 2500 --bSize 4
```

**Comment by Night1099** [comment]
any ideas or just need more data?

also could you explain what Kesi and  pix indicate?

**Comment by CGLiWang** [comment]
My paper has the detailed description about these.
